packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,601 @@
+const SideNavItemContent = ({
+  id,
+  label,
+  href,
+  target,
+  startIcon,
+  severity,
+  statusLabel,
+  endIcon,
+  onClick,
+  isSelected,
+  isDisabled,
+}: Pick<
+  SideNavItem,
+  | "id"
+  | "label"
+  | "href"
+  | "target"
+  | "startIcon"
+  | "severity"
+  | "statusLabel"
+  | "endIcon"
+  | "onClick"
+  | "isSelected"
+  | "isDisabled"
+>) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  return (
+    <SideNavListItemContainer
+      id={id}
+      key={id}
+      disabled={isDisabled}
+      aria-disabled={isDisabled}
+      isDisabled={isDisabled}
+      isSelected={isSelected}
+      odysseyDesignTokens={odysseyDesignTokens}
+    >
+      {
+        // Use Link for accessible nav items and div for disabled items
+        isDisabled ? (
+          <Box
+            sx={{
+              display: "flex",
+              alignItems: "center",
+              width: "100%",
+              minHeight: "45px",
+              padding: `${odysseyDesignTokens.Spacing3} ${odysseyDesignTokens.Spacing4}`,
+              color: `${odysseyDesignTokens.TypographyColorDisabled} !important`,
+            }}
+          >
+            <SideNavItemLinkContent
+              label={label}
+              startIcon={startIcon}
+              endIcon={endIcon}
+              statusLabel={statusLabel}
+              severity={severity}
+            />
+          </Box>
+        ) : (
+          <Link href={href || ""} target={target} onClick={onClick}>
+            <SideNavItemLinkContent
+              label={label}
+              startIcon={startIcon}
+              endIcon={endIcon}
+              statusLabel={statusLabel}
+              severity={severity}
+            />
+          </Link>
+        )
+      }
+    </SideNavListItemContainer>
+  );
+};
+
+export type SideNavProps = {
+  /**
+   * Side Nav header text that is usually reserved to show the App name
+   */
+  navHeaderText: string;
+  /**
+   *  Determines whether the side nav is collapsible
+   */
+  isCollapsible?: boolean;
+  /**
+   * Footer items in the side nav
+   */
+  footerItems?: SideNavFooterItem[];
+  /**
+   *  Triggers when the side nav is collapsed
+   */
+  onCollapse?(): void;
+  /**
+   * Nav items in the side nav
+   */
+  sideNavItems: SideNavItem[];
+} & Pick<HtmlProps, "testId">;
+
+const SideNav = ({
+  navHeaderText,
+  isCollapsible,
+  onCollapse,
+  sideNavItems,
+  footerItems,
+}: SideNavProps) => {
+  const [isSideNavCollapsed, setSideNavCollapsed] = useState(false);
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const processedSideNavItems = useMemo(
+    () =>
+      sideNavItems.map((item) => ({
+        ...item,
+        children: item.children?.map((childProps) => (
+          <SideNavItemContent key={childProps.id} {...childProps} />
+        )),
+      })),
+    [sideNavItems],
+  );
+
+  const sideNavCollapseHandler = useCallback(() => {
+    setSideNavCollapsed(!isSideNavCollapsed);
+    onCollapse && onCollapse();
```

Comment:
We can do this instead and be more clear:
```suggestion
    onCollapse?.();
```